Makefile

```diff
@@ -0,0 +1,34 @@
+# Get protoc commands
+PROTOC := $(shell command -v protoc)
+PROTOC_GEN_GO := $(shell command -v protoc-gen-go)
+PROTOC_GEN_GO_GRPC := $(shell command -v protoc-gen-go-grpc)
+
+# Get package manager
+PACKAGE_MGR := $(shell ([ -f /etc/redhat-release ] && echo yum) || ([ -f /etc/debian_version ] && echo apt-get))
+
+all: checkenv build
+
+build: checkenv
+	protoc --go_out=lib/go/ *.proto
+	#protoc --go_out=lib/go --go_opt=paths=source_relative *.proto
+	#protoc --go-grpc_out=lib/go --go-grpc_opt=paths=source_relative *.proto
+
+checkenv:
+	mkdir -p google/protobuf
+	wget https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/protoc-3.14.0-linux-x86_64.zip
+	unzip -j -d google/protobuf protoc-3.14.0-linux-x86_64.zip include/google/protobuf/descriptor.proto
+ifndef PROTOC
+	@echo "Installing Protoc"
+	sudo $(PACKAGE_MGR) install -y protobuf-compiler
+endif
+ifndef PROTOC_GEN_GO
+	@echo "Installing protoc-gen-go"
+	go get google.golang.org/protobuf/cmd/protoc-gen-go
+endif
+ifndef PROTOC_GEN_GO_GRPC
+	@echo "Installing protoc-gen-go-grpc"
+	go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
+endif
+
+clean:
+	rm -rf google/ lib/go/replication.pb.go lib/go/replication_grpc.pb.go protoc-3.14.0-linux-x86_64.zip
```

Comment:
remove data in /lib/go/replication/* folder

Reply:
Pls refer new one